sdv/multi_table/base.py

```diff
@@ -224,19 +224,22 @@ def validate(self, data):
                 A dictionary of table names to pd.DataFrames.
         """
         errors = []
-        try:
-            self.metadata.validate_data(data)
-        except InvalidDataError as e:
-            errors += e.errors
-
+        constraints_errors = []
+        self.metadata.validate_data(data)
         for table_name in data:
             if table_name in self._table_synthesizers:
-                errors += self._table_synthesizers[table_name]._validate_constraints(
-                    data[table_name])
+                try:
+                    self._table_synthesizers[table_name]._validate_constraints(data[table_name])
+                except ConstraintsNotMetError as error:
+                    constraints_errors.append(error)
+
                 # Validate rules specific to each synthesizer
                 errors += self._table_synthesizers[table_name]._validate(data[table_name])
 
-        if errors:
+        if constraints_errors:
+            raise ConstraintsNotMetError(constraints_errors)
+
+        elif errors:
             raise InvalidDataError(errors)
```

Comment:
Do we have a test that enters this code block?

Reply:
Now we do :)